### PR TITLE
Crud interface

### LINKED ISSUE #46: [Backend | Database Setup] Create Generic CRUD Database Gateway Interface

### PR DESCRIPTION
**Pull request by Jacob4649**
Closes #46. Specifies a CRUD interface for simple database interactions, also add Dynamo authentication to Github Actions test workflows for eventual Dynamo interacting tests. Depends on #40, cannot be merged before this is closed.

### CODE REVIEW

File: backend/src/main/java/com/javahelp/backend/data/ICRUDGateway.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.javahelp.backend.data;
+
+/**
+ * Interface specifying generic CRUD operations for the provided type.
+ * <p></p>
```
Comment (Lam207) [inline backend/src/main/java/com/javahelp/backend/data/ICRUDGateway.java:5]: Just a little question, what does ``<p></p>`` do, though?

**Review by Lam207 — APPROVED** [review]
Looks good to me! Can't find anything to improve.

File: backend/src/main/java/com/javahelp/backend/data/ICRUDGateway.java (hunk 1 above)
Comment (Jacob4649) [reply backend/src/main/java/com/javahelp/backend/data/ICRUDGateway.java:5]: It's just a new paragraph tag

**Review by yuanshu777 — APPROVED** [review]
I'm not familiar with the DynamoDB database, But this looks good to me

**Review by Enkhzul001 — APPROVED** [review]
Looks good.

**Pull request merged**